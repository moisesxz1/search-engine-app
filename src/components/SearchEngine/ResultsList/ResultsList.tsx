import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import ResultItem from '../ResultItem/ResultItem';

import Box from '@mui/material/Box';
import List from '@mui/material/List';
import Divider from '@mui/material/Divider';
import Alert from '@mui/material/Alert';

interface ResultsListProps {
  title: string;
  results: string[];
}

const ResultsList: React.FC<ResultsListProps> = (props) => {
  const { title, results } = props;

  return (
    <Fragment>
      <h2>{title}</h2>
      <Box>
        <Divider />
        <nav aria-label='secondary mailbox folders'>
          <List>
            {results.length > 0 ? (
              results.map((result: string) => <ResultItem key={result} name={result} />)
            ) : (
              <Alert severity='info'>No results were found</Alert>
            )}
          </List>
        </nav>
      </Box>
    </Fragment>
  );
};

ResultsList.propTypes = {
  title: PropTypes.string.isRequired,
  results: PropTypes.array.isRequired,
};

export default ResultsList;
