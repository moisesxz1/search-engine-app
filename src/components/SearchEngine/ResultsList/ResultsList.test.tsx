import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import ResultsList from './ResultsList';

describe('Tests on the ResultsList /> Component', () => {
  test('It should render <ResultsList /> correctly', () => {
    const wrapper = shallow(
      <ResultsList
        title='Google Results'
        results={[
          'Cereal & Granola - Walmart.com',
          'Cereal - Wikipedia',
          'How to Choose a Breakfast Cereal - Verywell Fit',
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('It should render the title prop  correctly', () => {
    const title = 'Google Results';

    const wrapper = shallow(
      <ResultsList
        title={title}
        results={[
          'Cereal & Granola - Walmart.com',
          'Cereal - Wikipedia',
          'How to Choose a Breakfast Cereal - Verywell Fit',
        ]}
      />
    );

    const titleProp = wrapper.find('h2').text();

    expect(titleProp).toBe(title);
  });

  test('It should render the results prop  correctly', () => {
    const title = 'Google Results';
    const results = [
      'Cereal & Granola - Walmart.com',
      'Cereal - Wikipedia',
      'How to Choose a Breakfast Cereal - Verywell Fit',
    ];

    const wrapper = shallow(<ResultsList title={title} results={results} />);

    expect(wrapper.find('ResultItem').length).toBe(results.length);
  });
});
