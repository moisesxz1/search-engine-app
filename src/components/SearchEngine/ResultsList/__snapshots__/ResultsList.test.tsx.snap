// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the ResultsList /> Component It should render <ResultsList /> correctly 1`] = `
<Fragment>
  <h2>
    Google Results
  </h2>
  <ForwardRef(Box)>
    <ForwardRef(Divider) />
    <nav
      aria-label="secondary mailbox folders"
    >
      <ForwardRef(List)>
        <ResultItem
          key="Cereal & Granola - Walmart.com"
          name="Cereal & Granola - Walmart.com"
        />
        <ResultItem
          key="Cereal - Wikipedia"
          name="Cereal - Wikipedia"
        />
        <ResultItem
          key="How to Choose a Breakfast Cereal - Verywell Fit"
          name="How to Choose a Breakfast Cereal - Verywell Fit"
        />
      </ForwardRef(List)>
    </nav>
  </ForwardRef(Box)>
</Fragment>
`;
